jenkins-job.sh

```diff
@@ -453,6 +465,139 @@ function run_release {
+
+    repo sync -j16 --force-sync -d -c
+
+    rm -rf ${RESULT_DIR}
+    mkdir -p ${RESULT_DIR}
+
+    rm -rf ${BUILD_DIR}/out
+
+    if [[ "${BUILD_VERSION}" = "5.1" ]] ; then
+        halium_build_device tenderloin cm_tenderloin-userdebug
+        halium_build_device mako aosp_mako-userdebug
+        halium_build_device hammerhead aosp_hammerhead-userdebug
+    else
+        halium_build_device onyx lineage_onyx-userdebug
+        halium_build_device mido lineage_mido-userdebug
+        halium_build_device rosy lineage_rosy-userdebug
+        halium_build_device athene lineage_athene-userdebug
+        halium_build_device tissot lineage_tissot-userdebug
+    fi
+}
+
+halium_publish_archive() {
+    archive=$1
+    echo "Publishing archive '$archive' ..."
+    mv $archive ${RESULT_DIR}
+}
+
+halium_generate_checksums() {
+    file=$1
+    echo "Generating checksums for archive '$file' ..."
+    md5sum $file > $file.md5sum
+    sha256sum $file > $file.sha256sum
+}
+
+halium_build_device() {
+    MACHINE=$1
+    BUILD_TARGET=$2
+    BUILD_CMD=lunch
+    [[ "${BUILD_VERSION}" = "7.1" ]] && BUILD_CMD=breakfast
+    OUTPUT_DIR=${BUILD_DIR}/out/target/product/${MACHINE}
+    ARCHIVE_NAME="${BASE_ARCHIVE_NAME}-${MACHINE}.tar.bz2"
+    DEBUG_ARCHIVE_NAME="${BASE_ARCHIVE_NAME}-${MACHINE}-dbg.tar.bz2"
+    KERNEL_PARTS_ARCHIVE_NAME="${BASE_ARCHIVE_NAME}-kernel-parts-${MACHINE}.tar.bz2"
+
+    echo "==============================================================="
+    echo "Machine: ${MACHINE}"
+    echo "Build version: ${BUILD_VERSION}"
+    echo "Build dir: ${BUILD_DIR}"
+    echo "Result dir: ${RESULT_DIR}"
+    echo "Output dir: ${OUTPUT_DIR}"
+    echo "Archive name: ${ARCHIVE_NAME}"
+    echo "==============================================================="
+
+    git config --global user.name "Jenkins"
+    git config --global user.email "jenkins@nas-admin.org"
+
+    [[ "${BUILD_VERSION}" = "5.1" ]] || mkdir -p ${BUILD_DIR}/out/host/linux-x86/framework/
+
+    cd ${BUILD_DIR}
+
+    # cleanup previous changes made by halium's device setup
+    repo forall -vc "git reset --hard"
+    # retrieve device's manifest
+    ./halium/devices/setup $MACHINE --force-sync
+
+    source build/envsetup.sh
+
+    #For Ubuntu 18.04 we will need to use either of below:
+    [[ "${BUILD_VERSION}" = "5.1" ]] || export LC_ALL=C
+    [[ "${BUILD_VERSION}" = "5.1" ]] && export USE_HOST_LEX=yes
+
+    export USE_CCACHE=1
+    #make clobber
+
+    ${BUILD_CMD} ${BUILD_TARGET}
+    mka systemimage
+    if [ $? != 0 ]; then
+        echo "Build of Halium ${BUILD_VERSION} for $MACHINE failed"
+        exit 1
+    fi
+
+    # Package result
+    cd ${OUTPUT_DIR}
+        #touch filesystem_config.txt
+        #cp ramdisk-android.img android-ramdisk.img
+        #tar cvjf ${ARCHIVE_NAME} system android-ramdisk.img filesystem_config.txt system.img
+        tar cvjf ${ARCHIVE_NAME} system.img
+        [[ "${BUILD_VERSION}" = "5.1" ]] && tar cvjf ${DEBUG_ARCHIVE_NAME} symbols/
```

Comment:
The idea of the "symbols" archive isn't specific to 5.1, and can very well be applied to 7.1